src/engine/wal/manager/builder.rs

```
@@ -86,14 +86,13 @@ impl<'a> WALBuilder<'a> {
-                    decoder.decode(&content).map_err(|e| {
-                        WALError::wrap(format!(
-                            "failed to decode log file {:?}: {}",
-                            log_path,
-                            e.to_string()
-                        ))
-                    })?;
+                let saved_entries: Vec<WALEntry> = decoder.decode(&content).map_err(|e| {
+                    WALError::wrap(format!(
+                        "failed to decode log file {:?}: {}",
+                        log_path,
+                        e.to_string()
```

Clippy (1.96.0)

[warning] `to_string` applied to a type that implements `Display` in `format!` args

(warning)